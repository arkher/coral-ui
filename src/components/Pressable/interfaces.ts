import React from 'react';
import { PressableProps } from 'react-native';
import { CustomBoxProps } from '../Box/interface';

export type CustomPressableProps = PressableProps &
  Partial<{
    children: React.ReactNode;
  }> &
  CustomBoxProps;
