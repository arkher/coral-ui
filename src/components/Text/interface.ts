import { TextProps } from '@shopify/restyle';
import { ReactNode } from 'react';
import { Theme } from '../../themes/institucional';

type CustomTextProps = Partial<{
  children?: ReactNode;
  fs?: Custom.FontSize;
  lh?: Custom.LineHeight;
  ls?: Custom.LetterSpacing;
  testID?: string;
}> &
  Partial<TextProps<Theme>>;

export { CustomTextProps };
