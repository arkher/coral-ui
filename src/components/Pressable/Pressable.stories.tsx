import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { withDesign } from 'storybook-addon-designs';
import { StoriesView } from '../../stories/StorieView';
import { COLORS } from '../../themes/tokens';
import Box from '../Box';
import Pressable from './Pressable';

type Props = {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  bg: any;
};

export default {
  title: 'Elements/Pressable',
  component: Pressable,
  argTypes: {
    bg: {
      options: COLORS,
      control: { type: 'select' },
    },
  },
  decorators: [withDesign],
};

const urlHandoff =
  'https://www.figma.com/file/3raVfIADTUZCzFOOaQ9PMQ/HANDOFF-%7C-Core-Components-Mobile-%7C-Institucional?node-id=203%3A1636';

// Stories
export const BasicPressable = ({ bg }: Props): React.ReactNode => (
  <Pressable
    flexDirection="row"
    justifyContent="space-between"
    alignItems="baseline"
    bg="feedback-info-base"
  >
    <Box width={100} height={100} bg="black" />
    <Box width={100} height={100} ml="md" bg="black" />
  </Pressable>
);

BasicPressable.parameters = {
  design: {
    type: 'figma',
    url: urlHandoff,
  },
};

// Add all stories to RN/Expo storybook
storiesOf('Pressable-ds', module)
  .addDecorator(getStory => <StoriesView>{getStory()}</StoriesView>)
  .add('Default', () => (
    <>
      <Text mb="sm">Pressable </Text>

      <Pressable boxProps={{ bg: 'white' }}>
        <Text p="md">Clique-me!</Text>
      </Pressable>
    </>
  ));
