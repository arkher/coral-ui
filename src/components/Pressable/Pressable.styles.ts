import { useTheme } from '@shopify/restyle';
import { ViewStyle, StyleSheet } from 'react-native';
import { shadowOffsetValues, shadowRadiusValues, Theme } from '../../themes';

interface Styles {
  customStyle: ViewStyle;
}

const useStyles = (): StyleSheet.NamedStyles<Styles> => {
  const { colors, borderRadii } = useTheme<Theme>();
  return StyleSheet.create({
    customStyle: {
      shadowOffset: shadowOffsetValues.lg,
      shadowColor: colors.black,
      shadowRadius: shadowRadiusValues.sm,
      shadowOpacity: 0.3,
      backgroundColor: colors.white,
      minHeight: 'auto',
      minWidth: 'auto',
      padding: 0,
      borderRadius: borderRadii.md,
      width: '100%',
    },
  });
};

export default useStyles;
