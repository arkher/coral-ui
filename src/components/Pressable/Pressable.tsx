/* eslint-disable camelcase */
import { createRestyleComponent, useTheme } from '@shopify/restyle';
import React from 'react';
import { Pressable, ViewStyle } from 'react-native';
import {
  borderRadius,
  borderWidth,
  heightComponent,
  opacity,
  shadowOffset,
  shadowOpacity,
  shadowRadius,
  Theme,
} from '../../themes';
import Box from '../Box';
import { CustomPressableProps } from './interfaces';
import useStyles from './Pressable.styles';

const CustomPressable: React.FC<CustomPressableProps> = ({
  children,
  style,
  testID,
  onPress,
  onPressIn,
  onPressOut,
  onLongPress,
  onBlur,
  onFocus,
  delayLongPress,
  disabled,
  hitSlop,
  pressRetentionOffset,
  android_disableSound,
  android_ripple,
  testOnly_pressed,
  p,
  pb,
  pl,
  pr,
  pt,
  flexDirection,
  justifyContent,
  alignItems,
  ...props
}: CustomPressableProps) => {
  const styles = useStyles();
  const { spacing } = useTheme<Theme>();
  const pressableProps = {
    onPress,
    onPressIn,
    onPressOut,
    onLongPress,
    onBlur,
    onFocus,
    delayLongPress,
    disabled,
    hitSlop,
    pressRetentionOffset,
    android_disableSound,
    android_ripple,
    testOnly_pressed,
    flexDirection,
    justifyContent,
    alignItems,
  };

  return (
    <Box
      bg="white"
      minHeight="auto"
      minWidth="auto"
      br="md"
      width="100%"
      testID="pressable-box"
      {...props}
    >
      <Pressable
        testID={testID || 'pressable'}
        style={({ pressed }) => [
          {
            elevation: pressed ? 1 : 3,
          },
          {
            padding: spacing[p as 'quark' | 'nano' | 'xs' | 'sm' | 'md' | 'lg'],
            paddingTop:
              spacing[pt as 'quark' | 'nano' | 'xs' | 'sm' | 'md' | 'lg'],
            paddingBottom:
              spacing[pb as 'quark' | 'nano' | 'xs' | 'sm' | 'md' | 'lg'],
            paddingRight:
              spacing[pr as 'quark' | 'nano' | 'xs' | 'sm' | 'md' | 'lg'],
            paddingLeft:
              spacing[pl as 'quark' | 'nano' | 'xs' | 'sm' | 'md' | 'lg'],
            flexDirection: flexDirection as
              | 'row'
              | 'column'
              | 'row-reverse'
              | 'column-reverse',
            justifyContent: justifyContent as 'center',
            alignItems: alignItems as 'center',
          },
          { ...(styles.customStyle as ViewStyle) },
          { ...(style as ViewStyle) },
        ]}
        {...pressableProps}
      >
        {children}
      </Pressable>
    </Box>
  );
};

export default createRestyleComponent<CustomPressableProps, Theme>(
  [
    borderRadius,
    borderWidth,
    opacity,
    shadowOffset,
    shadowRadius,
    shadowOpacity,
    heightComponent,
  ],
  CustomPressable,
);
