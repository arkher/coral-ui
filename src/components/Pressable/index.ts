export { default } from './Pressable';
export * from './Pressable';
