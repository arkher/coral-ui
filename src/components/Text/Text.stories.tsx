import React from 'react';
import { storiesOf } from '@storybook/react-native';
import Text from './Text';
// import { fontSizes, letterSpacings, lineHeights } from '../../themes/default';

export default {
  title: 'Elements/Text',
  component: Text,
};

// Stories
// eslint-disable-next-line arrow-body-style
export const BasicText = (): React.ReactNode => {
  return (
    <Text m="sm">
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Reiciendis rerum
      esse modi natus laborum dolorum quidem omnis alias autem! Beatae ut quo
      magnam rerum soluta labore laudantium? Quasi, molestias et!
    </Text>
  );
};

// Add all stories to RN/Expo storybook
storiesOf('Text', module).add('Basic', BasicText);
