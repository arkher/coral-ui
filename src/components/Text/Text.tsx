import React from 'react';
import { createRestyleComponent, createText } from '@shopify/restyle';
import {
  fontSize,
  letterSpacing,
  lineHeight,
  Theme,
} from '../../themes/default';
import { CustomTextProps } from './interface';

const Text = createText<Theme>();

const CustomText: React.FC<CustomTextProps> = ({ children, ...props }) => (
  <Text variant="regular" {...props}>
    {children}
  </Text>
);

export default createRestyleComponent<CustomTextProps, Theme>(
  [fontSize, lineHeight, letterSpacing],
  CustomText,
);
