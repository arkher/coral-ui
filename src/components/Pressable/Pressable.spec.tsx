import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react-native';
import { ThemeProvider } from '@shopify/restyle';
import Pressable from './Pressable';
import Text from '../Text';
import themeMaestro from '../../themes/maestro';

afterEach(cleanup);

jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon');

test('should execute press event function', () => {
  const onEventMock = jest.fn();
  const { getByTestId } = render(
    <ThemeProvider theme={themeMaestro}>
      <Pressable onPress={onEventMock}>
        <Text>Pressable</Text>
      </Pressable>
    </ThemeProvider>,
  );

  fireEvent.press(getByTestId('pressable'));
  expect(onEventMock).toHaveBeenCalledTimes(1);
});

test('should be black', () => {
  const onEventMock = jest.fn();
  const { getByTestId } = render(
    <ThemeProvider theme={themeMaestro}>
      <Pressable onPress={onEventMock} bg="black">
        <Text>Pressable</Text>
      </Pressable>
    </ThemeProvider>,
  );

  expect(getByTestId('pressable-box').props.style[0].backgroundColor).toBe(
    '#000000',
  );
});

test('should render correctly', async () => {
  const { toJSON } = render(
    <ThemeProvider theme={themeMaestro}>
      <Pressable onPress={() => undefined}>
        <Text>Label Pressable</Text>
      </Pressable>
    </ThemeProvider>,
  );

  expect(toJSON()).toMatchSnapshot();
});
