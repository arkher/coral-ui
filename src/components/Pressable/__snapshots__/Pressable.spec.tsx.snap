// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "marginBottom": 8,
        "marginLeft": 16,
        "marginRight": 16,
      },
      Array [
        Object {},
      ],
    ]
  }
  testID="pressable-box"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "elevation": 3,
        },
        Object {},
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
          "minHeight": "auto",
          "minWidth": "auto",
          "padding": 0,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 16,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 8,
          "width": "100%",
        },
      ]
    }
    testID="pressable"
  >
    <Text
      style={
        Array [
          Object {
            "display": "flex",
            "fontFamily": "\\"Inter\\", sans-serif",
            "fontWeight": "400",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      Label Pressable
    </Text>
  </View>
</View>
`;
